sherpa/models/tests/test_parameter.py

```diff
@@ -234,7 +238,135 @@ def func(a, b):
     uf = np.frompyfunc(func, nin=2, nout=1)
     p, p2 = setUp_composite()
     comp = uf(p, p2)
-    assert repr(comp) == "<BinaryOpParameter 'func(model.p, model.p2)'>"
+    assert repr(comp) == "<BinaryOpParameter 'func(model.p, (model.p2))'>"
+
+
+class TestBrackets:
+    """Provide a set of model instances for the tests."""
+
+    a = Parameter('m', 'a', 2)
+    b = Parameter('m', 'b', 3)
+    c = Parameter('m', 'c', 4)
+    d = Parameter('m', 'd', 5)
+
+    # It would be nice to instead use a principled set of states,
+    # but let's just try a somewhat-random set of expressions.
```

Comment:
If we ever add [hypothesis](https://hypothesis.readthedocs.io/en/latest/) to Sherpa testing, we can go for it ;-)